### PR TITLE
Feature/86cw4nf56 db helper functions

### PR DESCRIPTION
**Pull request by NikkiShao**
Created helper functions to get:
- service data (multiple) - with it cover image, and artist data aggregated
- service data (single) - with ALL image, artist data, and artist profile image
- booking data (multiple) - with its service data, COVER service image
- booking data (single) - with their service and user information
- user data (multiple) - with their profile image
- user data (single) - with their profile image

### CODE REVIEW

File: behind-the-veil-siteroot/imports/ui/components/DatabaseHelper.jsx [hunk 1]
```diff
@@ -86,19 +84,243 @@ export function getServices(service_publication, params, filter, requireArtist =
...
+        for (let j = 0; j < imagesData.length; j++) {
+            // find matching image for the service
+            if (imagesData[j].imageType === "service" && bookingsData[i].serviceId === imagesData[j].target_id) {
+                bookingsData[i].serviceImageData = imagesData[j].imageData;
+                break;
+            }
+        }
+    }
+
+    console.log("bookingsData", bookingsData)
```
Comment (joshualoongwy) [inline behind-the-veil-siteroot/imports/ui/components/DatabaseHelper.jsx:142]: Just wondering, are the console logs included in this PR included on purpose?

File: behind-the-veil-siteroot/imports/ui/components/DatabaseHelper.jsx [hunk 2]
```diff
@@ -86,19 +84,243 @@ export function getServices(service_publication, params, filter, requireArtist =
...
+                {"target_id": serviceArtistUsername}
+            ]
+        }).fetch()[0];
+    });
+
+    const isLoading = isLoadingService() || isLoadingArtist() || isLoadingServiceImages() || isLoadingArtistProfile();
+
+    console.log("service data", [serviceData, artistData, serviceImagesData, profileImageData])
+
+    return [isLoading, serviceData, artistData, serviceImagesData, profileImageData]
```
Comment (joshualoongwy) [inline behind-the-veil-siteroot/imports/ui/components/DatabaseHelper.jsx:240]: Thoughts on making each of these helpers return an object instead of an array? i.e. 
{isLoading, serviceData, artistData, serviceImagesData, profileImageData}

As the array gets bigger it becomes more difficult for the consumer of the function to retrieve the correct values because they'll need to care about the order of the return values

File: behind-the-veil-siteroot/imports/ui/components/DatabaseHelper.jsx
Comment (joshualoongwy) [inline behind-the-veil-siteroot/imports/ui/components/DatabaseHelper.jsx:1]: These functions look really helpful (I'll take a closer look soon), but one thing I wanted to point out is that since each of these functions use hooks (i.e. useSubscribe, useTracker), they are technically custom hooks themselves. 

If you're not familiar with hooks, here's a reference: https://react.dev/learn/reusing-logic-with-custom-hooks#extracting-your-own-custom-hook-from-a-component

but basically, hooks are functions that use react's inbuilt hooks (e.g. useState) or other custom hooks. A key difference between hooks and regular js functions is that hooks must be called at the top level of components to comply with react's rules (basically, don't call hooks conditionally, which we aren't doing here so all good). Another thing, by convention, hooks are prefixed with "use" to convey that they are react hooks.

TLDR: i suggest renaming the database helper functions so that they start with "use" (e.g. "useDatabaseBookings") b/c they are technically all custom hooks

File: behind-the-veil-siteroot/imports/ui/components/DatabaseHelper.jsx (hunk 2 above)
Comment (NikkiShao) [reply behind-the-veil-siteroot/imports/ui/components/DatabaseHelper.jsx]: makes sense!

File: behind-the-veil-siteroot/imports/ui/components/DatabaseHelper.jsx (hunk 1 above)
Comment (NikkiShao) [reply behind-the-veil-siteroot/imports/ui/components/DatabaseHelper.jsx]: removed all console logs

File: behind-the-veil-siteroot/imports/ui/components/DatabaseHelper.jsx
Comment (NikkiShao) [reply behind-the-veil-siteroot/imports/ui/components/DatabaseHelper.jsx:1]: thanks for letting me know! changed it and will take note in the future!

**Review by joshualoongwy — APPROVED** [review]
